stable/mongodb-replicaset/init/on-start.sh

```diff
@@ -128,6 +128,17 @@ for peer in "${peers[@]}"; do
 
         log '✓ Replica reached SECONDARY state.'
 
+        # create the metric user if it does not exist
```

Comment:
What I'm not understanding is how you're creating users from a `Secondary`. It should require the users to be created from the `Primary` that then gets replicated to the `Secondary`. Which is why Admin users don't need to be created in this section either and why I originally only had the metrics user code here:
https://github.com/helm/charts/blob/master/stable/mongodb-replicaset/init/on-start.sh#L154

I'll test this out on my cluster to check the behavior.

Reply:
Ahh I didn't see the `--host $peer` flag.  I tested it out on my cluster and can see how the upgrades were a problem. Overall it looks good, however I think to prevent this from being redundant and somewhat duplicate code maybe it can be abstracted out to a function that gets called by the `Primary` and `Secondary` when metrics is enabled.

Reply:
Cool.  @ssalaues do you want me to look into that and add this into this PR or should this be another PR?

Reply:
@sekka1 I would say add it to this PR but TBH it's all I could come up with so far to handle fresh installs and upgrades cleanly. It's possible @unguiculus has a better solution?

Reply:
@ssalaues I dont know how to make that better either.  It seems it is not much code repeat and this shouldnt grow much more (fingers crossed).

Reply:
@ssalaues @unguiculus can we merge this?